contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -8402,6 +8232,149 @@ rewrite_function_call_geospatial_func_ref_args(T ctx)
+{
+	/* Handles rewrite of geospatial function calls */
+	if (ctx->spatial_proc_name_server_database_schema())
+	{
+		if (ctx->spatial_proc_name_server_database_schema()->schema) throw PGErrorWrapperException(ERROR, ERRCODE_FEATURE_NOT_SUPPORTED, "Remote procedure/function reference with 4-part object name is not currently supported in Babelfish", getLineAndPos(ctx));
+
+		/* This if-elseIf clause rewrites the query in case of Geospatial function Call */
+		if (ctx->spatial_proc_name_server_database_schema()->geospatial_func_arg() && ctx->function_arg_list())
+			rewrite_function_call_geospatial_func_ref_args(ctx);
+		else if (ctx->spatial_proc_name_server_database_schema()->geospatial_func_no_arg() && !ctx->function_arg_list())
+			rewrite_function_call_geospatial_func_ref_no_arg(ctx);
+	}
+}
+
+static void
+handleGeospatialFunctionsInFuncCallExpr(TSqlParser::Func_call_exprContext *ctx)
+{
+	if(ctx != NULL && !ctx->DOT().empty())
+	{
+		std::vector<TSqlParser::Method_callContext *> method_calls = ctx->method_call();
+		for (size_t i = 0; i < method_calls.size(); ++i)
+		{
+			TSqlParser::Method_callContext *method = method_calls[i];
+			/* rewriting the query in case of Geospatial function Call -> function_call.method_call */
+			if(method->spatial_methods())
+			{
+				size_t ind;
+				if (i == 0) ind = ctx->function_call()->stop->getStopIndex();
+				else ind = method_calls[i-1]->stop->getStopIndex();
+				rewrite_geospatial_query_helper(ctx, method, ind);
+			}
+		}
+	}
+}
+
+static void
+handleGeospatialFunctionsInLocalIdExpr(TSqlParser::Local_id_exprContext *ctx)
+{
+	if(ctx != NULL && !ctx->DOT().empty())
+	{
+		std::vector<TSqlParser::Method_callContext *> method_calls = ctx->method_call();
+		for (size_t i = 0; i < method_calls.size(); ++i)
+		{
+			TSqlParser::Method_callContext *method = method_calls[i];
+			/* rewriting the query in case of Geospatial function Call -> local_id.method_call */
+			if(method->spatial_methods())
+			{
+				size_t ind;
+				if (i == 0) ind = ctx->local_id()->stop->getStopIndex();
+				else ind = method_calls[i-1]->stop->getStopIndex();
+				rewrite_geospatial_query_helper(ctx, method, ind);
+			}
+		}
+	}
+}
+
+static void
+handleGeospatialFunctionsInBracketExpr(TSqlParser::Bracket_exprContext *ctx)
+{
+	if(ctx != NULL && !ctx->DOT().empty())
+	{
+		std::vector<TSqlParser::Method_callContext *> method_calls = ctx->method_call();
+		for (size_t i = 0; i < method_calls.size(); ++i)
+		{
+			TSqlParser::Method_callContext *method = method_calls[i];
+			/* rewriting the query in case of Geospatial function Call -> LR_BRACKET expression RR_BRACKET.method_call */
+			if(method->spatial_methods())
+			{
+				size_t ind;
+				std::string context = ::getFullText(ctx);
+				size_t spaces = 0;
+				for (size_t x = ctx->expression()->stop->getStopIndex() + 1 - ctx->start->getStartIndex(); x <= ctx->stop->getStopIndex() - ctx->start->getStartIndex(); x++)
```

Comment:
What are we exactly doing here? Why cant we get the index of RR_BRACKET?

Reply:
Done